From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Pranav Vashi <neobuddy89@gmail.com>
Date: Fri, 6 Sep 2024 13:16:04 +0530
Subject: [PATCH 127/226] services: Disallow max cached processes above 128

* No QCOM devices has value above 128. Let's prevent devices misusing device config
  or overlay to modify no. of cached processes.
* Shorter time to kill empty app processes.

Signed-off-by: Pranav Vashi <neobuddy89@gmail.com>
---
 .../java/com/android/server/am/ActivityManagerConstants.java  | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/services/core/java/com/android/server/am/ActivityManagerConstants.java b/services/core/java/com/android/server/am/ActivityManagerConstants.java
index f1da8b80f0..ccc5ce282c 100644
--- a/services/core/java/com/android/server/am/ActivityManagerConstants.java
+++ b/services/core/java/com/android/server/am/ActivityManagerConstants.java
@@ -185,7 +185,7 @@ final class ActivityManagerConstants extends ContentObserver {
      */
     static final String KEY_FREEZER_CUTOFF_ADJ = "freezer_cutoff_adj";
 
-    private static final int DEFAULT_MAX_CACHED_PROCESSES = 1024;
+    private static final int DEFAULT_MAX_CACHED_PROCESSES = 128;
     private static final boolean DEFAULT_PRIORITIZE_ALARM_BROADCASTS = true;
     private static final long DEFAULT_FGSERVICE_MIN_SHOWN_TIME = 2*1000;
     private static final long DEFAULT_FGSERVICE_MIN_REPORT_TIME = 3*1000;
@@ -947,7 +947,7 @@ final class ActivityManagerConstants extends ContentObserver {
     private static final String KEY_MAX_EMPTY_TIME_MILLIS =
             "max_empty_time_millis";
 
-    private static final long DEFAULT_MAX_EMPTY_TIME_MILLIS = 1000L * 60L * 60L * 1000L;
+    private static final long DEFAULT_MAX_EMPTY_TIME_MILLIS = 30L * 60L * 1000L;
 
     volatile long mMaxEmptyTimeMillis = DEFAULT_MAX_EMPTY_TIME_MILLIS;
 
